gem/lib/earlgrey/configure_earlgrey_pods.rb

```diff
@@ -0,0 +1,388 @@
+
+      path = File.join(Dir.pwd, File.basename(xcode_file, '.*') + ext)
+      path ? path : nil
+    end
+
+    # Strips excessive spaces from a string
+    # @param string [String] the string to process
+    # @return [String] the modified string
+    def strip(string)
+      string.split("\n").map(&:strip).join("\n")
+    end
+
+    # Raise error message after removing excessive spaces.
+    # @param message [String] the message to raise
+    # @return [nil]
+    def error(message)
+      raise strip(message)
+    end
+
+    # Prints string as blue after stripping excess spacing
+    # @param string [String] the string to print
+    # @return [nil]
+    def puts_blue(string)
+      puts strip(string).blue
+    end
+
+    # Prints string as yellow after stripping excess spacing
+    # @param string [String] the string to print
+    # @return [nil]
+    def puts_yellow(string)
+      puts strip(string).yellow
+    end
+
+    # Main entry point. Configures An Xcode project for use with EarlGrey.
+    #
+    # @param installer [Installer] Cococapods installer object or nil.
+    # @param project_name [String] the xcodeproj file name
+    # @param test_target_name [String] the test target name contained in xcodeproj
+    # @param scheme_file [String] the scheme file name. defaults to project name when nil.
+    # @return [nil]
+    def configure_for_earlgrey(installer, project_name, test_target_name, scheme_file)
+      puts_blue "Checking and Updating #{project_name} for EarlGrey."
+      pods_project = installer ? installer.pods_project : true
+      project_file = path_for project_name, '.xcodeproj'
+
+      raise 'No test target provided' unless test_target_name
+
+      if pods_project.nil? || project_file.nil?
+        error <<-E
+          The target's xcodeproj file could not be found.
+          Please check if the correct PROJECT_NAME is being passed in the Podfile.
+          Current PROJECT_NAME is: #{project_name}
+        E
+      end
+
+      @project_name = project_name
+      @test_target_name = test_target_name
+      @scheme_file = File.basename(scheme_file, '.*') + '.xcscheme'
+      @user_project = Xcodeproj::Project.open(project_file)
+      all_targets = user_project.targets
+      @test_target = all_targets.find { |target| target.name == test_target_name }
+      raise "Unable to find target: #{test_target_name}. Targets are: #{all_targets.map(&:name)}" unless test_target
+
+      # Add a Test Action to the User Project Scheme.
+      scheme = modify_scheme_for_actions
+
+      # Add a Copy Files Build Phase for EarlGrey.framework to embed it into the app under test.
+      # carthage uses carthage copy-frameworks instead of a copy files build phase.
+      add_earlgrey_copy_files_script unless carthage
+
+      # Add header/framework search paths for carthage
+      add_carthage_search_paths
+
+      # Adds BridgingHeader.h, EarlGrey.swift and sets bridging header.
+      copy_swift_files
```

Comment:
This code is the earlgrey gem so it doesn't run unless the gem is installed. The method doesn't assume Swift is the default, the defaults are set in the Thor CLI file `cli.rb`. If Swift is false then `copy_swift_files` is a no-op.